platform/wayland/cog-platform-wl.c

```diff
@@ -477,9 +502,17 @@ registry_on_global(void *data, struct wl_registry *registry, uint32_t name, cons
     } else if (strcmp(interface, zwp_fullscreen_shell_v1_interface.name) == 0) {
         display->fshell = wl_registry_bind(registry, name, &zwp_fullscreen_shell_v1_interface, 1);
     } else if (strcmp(interface, wl_seat_interface.name) == 0) {
-        display->seat.seat_name = name;
-        display->seat.seat_version = MAX(3, MIN(version, 7));
-        display->seat.seat = wl_registry_bind(registry, name, &wl_seat_interface, MAX(3, MIN(version, 7)));
+        uint32_t        seat_version = MAX(3, MIN(version, 7));
+        struct wl_seat *wl_seat = wl_registry_bind(registry, name, &wl_seat_interface, seat_version);
+        CogWlSeat      *seat = cog_wl_seat_create(wl_seat, name);
+        cog_wl_display_add_seat(display, seat);
+        static const struct wl_seat_listener seat_listener = {
+            .capabilities = seat_on_capabilities,
+#ifdef WL_SEAT_NAME_SINCE_VERSION
+            .name = seat_on_name,
+#endif // WL_SEAT_NAME_SINCE_VERSION
+        };
+        wl_seat_add_listener(wl_seat, &seat_listener, seat);
```

Comment:
This comes from https://github.com/Igalia/cog/pull/623/commits/a321b21181571ea271b3465d181e2581e9bc38d7#diff-7342248ee4ad822cd294d9c8341a52fd10b3d1df4dd2a202ccb926df67d51034L1710 but the seat_listener nows passes the CogWlSeat instance  as context.